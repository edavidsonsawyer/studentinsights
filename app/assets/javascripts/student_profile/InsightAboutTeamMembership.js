import React from 'react';
import PropTypes from 'prop-types';
import InsightQuote from './InsightQuote';
import Team from '../components/Team';
import FitText from '../components/FitText';


// Render an insight about sports team membership
export default class InsightAboutTeamMembership extends React.Component {
  render() {
    const {insightPayload, firstName} = this.props;
    const team = insightPayload;
    const isOrWas = team.active ? 'is' : 'was';
    return (
      <InsightQuote
        className="InsightAboutTeamMembership"
        quoteEl={
          <FitText
            minFontSize={12}
            maxFontSize={42}
            fontSizeStep={6}
            text={<span>{firstName} {isOrWas} on the <Team team={team} /> team</span>}
          />
        }
        sourceEl={
          <div>
            <div>with coach {team.coach_text}</div>
            <div>during the {team.season_key} season</div>
          </div>
        }
      />
    );
  }
}
InsightAboutTeamMembership.propTypes = {
  firstName: PropTypes.string.isRequired,
  insightPayload: PropTypes.shape({
    activity_text: PropTypes.string.isRequired,
    season_key: PropTypes.string.isRequired,
    coach_text: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired
  }).isRequired
};


export const ABOUT_TEAM_MEMBERSHIP = 'about_team_membership';
