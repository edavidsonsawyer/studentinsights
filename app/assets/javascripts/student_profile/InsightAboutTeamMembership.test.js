import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withDefaultNowContext} from '../testing/NowContainer';
import InsightAboutTeamMembership from './InsightAboutTeamMembership';


function testProps(props = {}) {
  return {
    firstName: 'Mari',
    insightPayload: {
      activity_text: 'Indoor Track - Girls Varsity',
      season_key: 'winter',
      coach_text: '2018-19',
      active: true,
    },
    ...props
  };
}

function testEl(props = {}) {
  return withDefaultNowContext(<InsightAboutTeamMembership {...props} />);
}


// since element sizing methods aren't present in the Jest JS environment
jest.mock('../components/FitText', () => 'mocked-fit-text');

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(testEl(props), el);
});

it('snapshots view', () => {
  const props = testProps();
  const tree = renderer
    .create(testEl(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
