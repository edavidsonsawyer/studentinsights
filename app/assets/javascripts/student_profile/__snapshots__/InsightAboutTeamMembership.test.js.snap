// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  className="InsightQuote InsightAboutTeamMembership"
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "margin": 20,
        "marginBottom": 0,
        "marginTop": 15,
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "overflow": "hidden",
        }
      }
    >
      <mocked-fit-text
        fontSizeStep={6}
        maxFontSize={42}
        minFontSize={12}
        text={
          <span>
            Mari
             
            is
             on the 
            <Team
              team={
                Object {
                  "active": true,
                  "activity_text": "Indoor Track - Girls Varsity",
                  "coach_text": "2018-19",
                  "season_key": "winter",
                }
              }
            />
             team
          </span>
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 12,
        "justifyContent": "space-between",
        "margin": 20,
        "marginBottom": 15,
        "marginTop": 10,
      }
    }
  >
    <div>
      <div>
        <div>
          with coach 
          2018-19
        </div>
        <div>
          during the 
          winter
           season
        </div>
      </div>
    </div>
  </div>
</div>
`;
